internal/generator/vector/filter/openshift/viaq/v1/normalize.go

```diff
@@ -26,55 +26,95 @@ if ._internal.log_source == "syslog" && exists(._internal.structured) {. = merge
 if ._internal.log_source == "container" && exists(._internal.structured) {.structured = ._internal.structured }
 `
 	SetLogLevel = `
+
 if !exists(._internal.level) {
-  level = "default"
+  level = null
   message = ._internal.message
 
-  # Match on well known structured patterns
-  # Order: emergency, alert, critical, error, warn, notice, info, debug, trace
+  # attempt 1: parse as logfmt (e.g. level=error msg="Failed to connect")
+  
+  parsed_logfmt, err = parse_logfmt(message)  
+  if err == null && is_string(parsed_logfmt.level) {
+    level = downcase!(parsed_logfmt.level)
+  }
 
-  if match!(message, r'^EM[0-9]+|level=emergency|Value:emergency|"level":"emergency"') {
```

Comment:
I think this fallback would be better if we took the keyword that occurs _first_ in the log message, rather than the highest-priority one, e.g.
```
Apr 24 15:02:47 fedora gnome-session-binary[3056]: GnomeDesktop-WARNING: Could not create transient scope for PID 3303: GDBus.Error:org.freedesktop.DBus.Error.UnixProcessIdUnknown: Failed to set unit properties: No such process
```
This is a Warning with a message that contains the word Error. If you use the earliest keyword found "WARNING" you get this right, if you use the scariest keyword, "ERROR" you exaggerate the level.

Reply:
done